python_modules/dagster/dagster/core/execution/plan/inputs.py

```diff
@@ -486,13 +488,64 @@ def compute_version(
         pipeline_def: PipelineDefinition,
         resolved_run_config: ResolvedRunConfig,
     ) -> Optional[str]:
+        solid_config = resolved_run_config.solids.get(str(self.solid_handle))
+        config_data = solid_config.inputs.get(self.input_name) if solid_config else None
 
-        config_data = self.get_associated_config(resolved_run_config)
-        input_def = self.get_associated_input_def(pipeline_def)
-        dagster_type = input_def.dagster_type
+        solid_def = pipeline_def.get_solid(self.solid_handle)
+        dagster_type = solid_def.input_def_named(self.input_name).dagster_type
         return dagster_type.loader.compute_loaded_input_version(config_data)
 
 
+@whitelist_for_serdes
+class FromDirectInputValue(
+    NamedTuple(
+        "_FromDirectInputValue",
+        [("input_name", str)],
+    ),
+    StepInputSource,
+):
+    """This input source is for direct python values to be passed as inputs to ops."""
+
+    def __new__(cls, input_name: str):
+        return super(FromDirectInputValue, cls).__new__(
+            cls,
+            input_name=input_name,
+        )
+
+    def get_associated_input_def(self, pipeline_def: PipelineDefinition) -> InputDefinition:
+        return pipeline_def.graph.input_def_named(self.input_name)
```

Comment:
whats this for?

Reply:
darn. thought this was also on the base class.